src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,145 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.UserInvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private static final int VALID_AGE = 21;
+    private static final String VALID_LOGIN = "validLogin";
+    private static final String VALID_PASSWORD = "validPassword";
+
+    @BeforeAll
+    static void beforeAll() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void testRegister_userWhoIsRegistered_notOk() {
+        User user = new User();
+        user.setAge(19);
+        user.setLogin("login1234");
+        user.setPassword("password1234");
+        storageDao.add(user);
+        assertThrows(UserInvalidDataException.class, () -> registrationService.register(user),
+                "method need throw UserInvalidDataException for user that contains in storage");
+    }
+
+    @Test
+    void testRegister_null_notOk() {
+        assertThrows(UserInvalidDataException.class, () -> registrationService.register(null),
+                "method need to throw UserInvalidDataException for user: null");
+    }
+
+    @Test
+    void testRegister_nullFields_not_Ok() {
+        User userNullLogin = new User();
+        userNullLogin.setPassword(VALID_PASSWORD);
+        User userNullPassword = new User();
+        userNullPassword.setLogin(VALID_LOGIN);
+        assertThrows(UserInvalidDataException.class,
+                () -> registrationService.register(userNullLogin),
+                "method need throw UserInvalidDataException for login: null");
+        assertThrows(UserInvalidDataException.class,
+                () -> registrationService.register(userNullPassword),
+                "method need throw UserInvalidDataException for password: null");
+    }
+
+    @Test
+    void testRegister_nonValidAge_notOk() {
+        User userInvalidAge = new User();
+        userInvalidAge.setLogin(VALID_LOGIN);
+        userInvalidAge.setPassword(VALID_PASSWORD);
+        userInvalidAge.setAge(10);
```

Comment:
It is better to move it to a constant

Reply:
i already created constants for valid age, i think it will be too much constant if i create it for all non valid fields